src/modules/audit/components/auditHistoryColumnDefs.tsx

```diff
@@ -0,0 +1,161 @@
+import { Link, Stack, Typography } from '@mui/material';
+import { compact, filter } from 'lodash-es';
+import {
+  ContextualCollapsibleList,
+  ContextualListExpansionButton,
+} from '@/components/elements/CollapsibleListContext';
+import { ColumnDef } from '@/components/elements/table/types';
+import AuditObjectChangesSummary, {
+  ObjectChangesType,
+} from '@/modules/audit/components/AuditObjectChangesSummary';
+import { hasMeaningfulValue } from '@/modules/form/util/formUtil';
+import RelativeDateTableCellContents from '@/modules/hmis/components/RelativeDateTableCellContents';
+import { auditActionForDisplay } from '@/modules/hmis/hmisUtil';
+import { Routes } from '@/routes/routes';
+import { AuditEventType } from '@/types/gqlTypes';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+export type AuditHistoryNode = NonNullable<{
+  id: string;
+  createdAt: string;
+  event: AuditEventType;
+  objectChanges?: any | null;
+  recordName: string;
+  graphqlType: string;
+  recordId: string;
+  user?: {
+    id: string;
+    name: string;
+  } | null;
+  trueUser?: {
+    id: string;
+    name: string;
+  } | null;
+  clientId?: string;
+  clientName?: string;
+  projectId?: string;
+  projectName?: string;
+  enrollmentId?: string;
+}>;
+
+export const auditHistoryColumns: ColumnDef<AuditHistoryNode>[] = [
+  {
+    key: 'timestamp',
+    header: 'Timestamp',
+    width: '180px',
+    render: (e) => (
+      <RelativeDateTableCellContents dateTimeString={e.createdAt} />
+    ),
+  },
+  {
+    key: 'user',
+    header: 'User',
+    width: '180px',
+    render: ({ user, trueUser }) =>
+      compact([trueUser?.name, user?.name]).join(' acting as '),
+  },
+  {
+    key: 'clientName',
+    header: 'Client Name',
+    width: '180px',
+    render: ({ clientName, clientId }) => {
+      if (!clientName) return;
+      if (clientName && clientId) {
+        return (
+          <Link
+            href={generateSafePath(Routes.CLIENT_DASHBOARD, {
+              clientId: clientId,
+            })}
+          >
+            {clientName}
+          </Link>
+        );
+      }
+      return clientName; // Should never get here, but just in case
+    },
+  },
+  {
+    key: 'projectName',
+    header: 'Project Name',
+    width: '180px',
+    render: ({ projectName, clientId, enrollmentId, projectId }) => {
+      if (!projectName) return;
+      if (clientId && enrollmentId) {
+        // Link to the enrollment
+        return (
+          <Link
+            href={generateSafePath(Routes.ENROLLMENT_DASHBOARD, {
```

Comment:
We always use RouterLink for internal links! that uses the react-router-dom component
```suggestion
          <RouterLink
            to={generateSafePath(Routes.ENROLLMENT_DASHBOARD, {
```